scripts/battle/GameState.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using TeicsoftSpectacleCards.scripts.XmlParsing;
+using TeicsoftSpectacleCards.scripts.XmlParsing.models;
+
+namespace TeicsoftSpectacleCards.scripts.battle;
+
+public class GameState
+{
+    public int ComboMultiplier { get; set; }
+    public int SpectaclePoints { get; set; }
+    public int MaxPlayerHealth { get; set; }
+    public int PlayerHealth { get; set; }
+    public List<Card> CardStack { get; set; } // changed this back to Card objects, as we use spectacle points in the combo processing. Easier than tracking separately.
+    private List<ComboModel> AllCombos { get; set; }
+
+    // Constructor
+    public GameState()
+    {
+        AllCombos = ComboXmlParser.ParseAllCombos(); // Retrieve a list of all combos as model objects
+
+        CardStack = new List<Card>();
+        ComboMultiplier = 1; // 1 is lowest possible value
+        SpectaclePoints = 0;
+        MaxPlayerHealth = 100; // adjust this as needed, or base on some other check
+        PlayerHealth = MaxPlayerHealth;
+    }
+    
+    // Stack Methods
+    public void PushCardStack(Card card)
+    {
+        CardStack.Add(card);
+    }
+
+    public void CleanCardStack()
+    {
+        CardStack = new List<Card>();
+
+    }
+    
+    // Player Health Methods
+    public bool IsPlayerAlive()
+    {
+        return PlayerHealth > 0;
+    }
+
+    public void AdjustHealth(int amount)
+    {
+        PlayerHealth += amount;
+
+        if (PlayerHealth > MaxPlayerHealth)
+        {
+            PlayerHealth = MaxPlayerHealth;
+        }
+        
+        if (!IsPlayerAlive())
+        {
+            //end round on loss
+        }
+    }
```

Comment:
Like when we discussed negative damage vs healing when it comes to enemies, we'll need to separate this out.

Reply:
Robbed some of your enemy code and did this